antsibull/docs_parsing/ansible_internal.py

```diff
@@ -0,0 +1,78 @@
+# Author: Felix Fontein <felix@fontein.de>
+# Author: Toshio Kuratomi <tkuratom@redhat.com>
+# License: GPLv3+
+# Copyright: Ansible Project, 2020
+"""Parse documentation from ansible plugins using anible-doc."""
+
+import json
+import tempfile
+import typing as t
+
+from ..logging import log
+from ..utils.get_pkg_data import get_antsibull_data
+from ..vendored.json_utils import _filter_non_json_lines
+from . import _get_environment
+
+if t.TYPE_CHECKING:
+    from ..venv import VenvRunner, FakeVenvRunner
+
+
+mlog = log.fields(mod=__name__)
+
+
+async def get_ansible_plugin_info(venv: t.Union['VenvRunner', 'FakeVenvRunner'],
+                                  collection_dir: t.Optional[str],
+                                  collection_names: t.Optional[t.List[str]] = None
+                                  ) -> t.Dict[str, t.Dict[str, t.Any]]:
+    """
+    Retrieve information about all of the Ansible Plugins.
+
+    :arg venv: A VenvRunner into which Ansible has been installed.
+    :arg collection_dir: Directory in which the collections have been installed.
+                         If ``None``, the collections are assumed to be in the current
+                         search path for Ansible.
+    :arg collection_names: Optional list of collections. If specified, will only collect
+                           information for plugins in these collections.
+    :returns: A nested directory structure that looks like::
+
+        plugin_type:
+            plugin_name:  # Includes namespace and collection.
+                {information from ansible-doc --json.  See the ansible-doc documentation for more
+                 info.}
+    """
+    flog = mlog.fields(func='get_ansible_plugin_info')
+    flog.debug('Enter')
+
+    env = _get_environment(collection_dir)
+
+    venv_python = venv.get_command('python')
+
+    with tempfile.NamedTemporaryFile() as tmp_file:
+        tmp_file.write(get_antsibull_data('collection-enum.py'))
```

Comment:
I was thinking we should have this run in-process instead of being a separate, internal script.  Is that something we can do?   Is the dependence on ansible-base code the reason it was easier to write this as a separate script at first?

Reply:
For the `stable` subcommand, we really must run it in the venv, otherwise it does not necessarily have access to the correct ansible-base version. (For the `current` subcommand it doesn't matter.)